diff --git a/CMakeLists.txt b/CMakeLists.txt
index eb50436..8f703b1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -175,7 +175,7 @@ if(NOT BOOST_SOURCE_DIR)
  if(NOT BOOST_MODULE_DIR)
   EXEC_PYTHON_SCRIPT ("import boost; print boost.__path__[0]" BOOST_MODULE_DIR)
  endif(NOT BOOST_MODULE_DIR)
- execute_process(COMMAND ln -fs ${BOOST_MODULE_DIR} ${CMAKE_CURRENT_BINARY_DIR}/pytriqs/)
+ execute_process(COMMAND ln -fs ${BOOST_MODULE_DIR} ${CMAKE_CURRENT_BINARY_DIR}/pytriqs/boost)
  install (DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/pytriqs/boost DESTINATION ${TRIQS_PYTHON_LIB_DEST})
 endif(NOT BOOST_SOURCE_DIR)
 
